byzcoin/service.go

```diff
@@ -1553,6 +1557,43 @@ func newService(c *onet.Context) (onet.Service, error) {
 		return nil, err
 	}
 
+	ver, err := s.LoadVersion()
+	if err != nil {
+		return nil, err
+	}
+	switch ver {
+	case 0:
+		// Version 0 means it hasn't been set yet. If there are any ByzCoin_[0-9af]+
+		// buckets, then they must be old format.
+		db, _ := s.GetAdditionalBucket([]byte("check-db-version"))
```

Comment:
we're using `s.GetAdditionalBucket` is because we cannot get a database pointer with any other way?

Reply:
yes.